BrightFutures/Future.swift

```diff
@@ -392,21 +545,63 @@ public extension Future {
 
         return p.future
     }
-    
+	
+	/**
+	Registers the given closure that provides a recovery value when this future fails.
+	
+	When this future fails, the given closure will be called, the result from that closure will
```

Comment:
I think this could be clearer. Suggestion: "Returns a future that, if this future succeeds, contains the value that this future succeeded with or, if this future fails, the value that is returned from the given task."